File: android/src/main/java/com/kakaotam/rnsdk/common/SdkInitializer.kt
package com.kakaotam.rnsdk.common

import com.facebook.react.bridge.ReactApplicationContext
import com.kakao.sdk.common.KakaoSdk

class SdkInitializer private constructor() {
    companion object {
        @JvmStatic
        var initialized = false;

        fun ensure(reactContext: ReactApplicationContext) {
            if (!initialized) {
                val appKey =
                    reactContext.resources.getString(
                        reactContext.resources.getIdentifier(
                            "KAKAO_NATIVE_APP_KEY",
                            "string",
                            reactContext.packageName
                        )
                    )
                if (appKey.isNotBlank()) {
                    KakaoSdk.init(
                        reactContext.currentActivity!!,
                        appKey,
                        null,
                        null,
                        null,
                        null
                    )
                    initialized = true;
                }
            }
        }
    }
}

File: SdkInitializer.swift
//
//  SdkInitializer.swift
//  react-native-kakao-sdk-all
//
//  Created by kakao-tam on 2022/05/06.
//

import Foundation

import KakaoSDKCommon

final class SdkInitializer {
    static let shared = SdkInitializer()
    
    var initialized: Bool = false
    
    private init() {}
    
    func ensure() {
        if !initialized {
            let appKey = Bundle.main.object(forInfoDictionaryKey: "KAKAO_NATIVE_APP_KEY") as? String
            if let appKey = appKey {
                KakaoSDK.initSDK(appKey: appKey)
                initialized = true
            }
        }
    }
}
